### PR TITLE
chore: allow codecov to detect SHA

### PR DESCRIPTION
**Pull request by mistercrunch**
In our CI, where codecov.sh is called, there's a message from its CLI saying `Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0`. From my understanding setting depth=2 should allow to resolve this, without having much impact on cloning workload.

### CODE REVIEW

**Review by eschutho — APPROVED** [review]

**Comment by mistercrunch** [comment]
Switching to https://github.com/codecov/codecov-action may be the best option, for now we just need to unblock this. Anyone looking at these reports?

File: .github/workflows/superset-python-integrationtest.yml [hunk 1]
```diff
@@ -67,6 +68,9 @@ jobs:
           ./scripts/python_tests.sh
       - name: Upload code coverage
         if: steps.check.outputs.python
+        continue-on-error: true # codecob is not critical and randomly fails at times
```
Comment (rusackas) [inline .github/workflows/superset-python-integrationtest.yml:71]:
```suggestion
        continue-on-error: true # codecov is not critical and randomly fails at times
```
Code on the Cob?! Sign me up!

This still upholds/blocks on the standards (e.g. core packages at 100% coverage) right?

File: .github/workflows/superset-frontend.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} )"
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0 # Fetch all history
+          fetch-depth: 2 # codecov needs 2 to get a read on the SHA
```
Comment (rusackas) [inline .github/workflows/superset-frontend.yml:23]: Cool... 0 worked as prescribed, but left me wondering if I was wasting resources.

File: .github/workflows/superset-python-integrationtest.yml [hunk 3]
```diff
@@ -124,6 +128,9 @@ jobs:
...
```
Comment (rusackas) [inline .github/workflows/superset-python-integrationtest.yml:131]:
```suggestion
        continue-on-error: true # codecov is not critical and randomly fails at times
```

File: .github/workflows/superset-python-integrationtest.yml [hunk 4]
```diff
@@ -172,5 +180,8 @@ jobs:
...
```
Comment (rusackas) [inline .github/workflows/superset-python-integrationtest.yml:183]:
```suggestion
        continue-on-error: true # codecov is not critical and randomly fails at times
```

File: .github/workflows/superset-python-presto-hive.yml [hunk 5]
```diff
@@ -139,5 +140,8 @@ jobs:
           ./scripts/python_tests.sh -m 'chart_data_flow or sql_json_flow'
       - name: Upload code coverage
         if: steps.check.outputs.python
+        continue-on-error: true # codecob is not critical and randomly fails at times
```
Comment (rusackas) [inline .github/workflows/superset-python-presto-hive.yml:143]:
```suggestion
        continue-on-error: true # codecov is not critical and randomly fails at times
```

File: .github/workflows/superset-python-unittest.yml [hunk 6]
```diff
@@ -47,5 +48,8 @@ jobs:
           pytest --durations-min=0.5 --cov-report= --cov=superset ./tests/common ./tests/unit_tests --cache-clear
       - name: Upload code coverage
         if: steps.check.outputs.python
+        continue-on-error: true # codecob is not critical and randomly fails at times
```
Comment (rusackas) [inline .github/workflows/superset-python-unittest.yml:51]:
```suggestion
        continue-on-error: true # codecov is not critical and randomly fails at times
```

**Comment by mistercrunch** [comment]
Ok I ended up setting up the official GHA, allowed me to delete that heavy `codecov.sh` @ 200 LoC

**Pull request merged**